### PR TITLE
#156372855 Get the current user profile

### PR DESCRIPTION
**Pull request by chadwalt**
#### What does this PR do?
Enables the logged in user to retrieve his/her profile information.

#### Description of Task to be completed?
When the profile endpoint is called, the logged in user should be able to view his/her profile information.

#### How should this be manually tested?
This can be tested by calling this endpoint in postman `localhost:8000/api/v1/user/profile`

#### What are the relevant pivotal tracker stories?
[156372855](https://www.pivotaltracker.com/story/show/156372855)

#### Any background context you want to add?
N/A
#### Screenshots
N/A

### CODE REVIEW

File: tests/App/Http/Controllers/V1/UserControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        $this->json("GET", "api/v1/user/profile")
+            ->seeJson(
+                [
+                    "first_name" => "Test",
+                    "last_name" => "User",
+                    "email" => "test.user@andela.com"
+                ]
+            );
+    }
+}
```
Comment (emugaya) [inline tests/App/Http/Controllers/V1/UserControllerTest.php]: Add newline

File: app/Http/Controllers/V1/UserController.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+use App\Models\Profile;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+
+/**
+ * Class UserController - Handles users operations.
+ *
+ * @package App\Http\Controllers\V1
+ */
+class UserController extends Controller
```
Comment (emugaya) [inline app/Http/Controllers/V1/UserController.php]: I suggest you change this controller name to Profile Controller.

File: app/Http/Controllers/V1/UserController.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+{
+    use RESTActions;
```
Comment (emugaya) [inline app/Http/Controllers/V1/UserController.php]: How is `RESTActions` being used in this controller?

File: app/Http/Controllers/V1/UserController.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+     *
+     * @param int $request - The request object.
+     *
+     * @return Response object.
+     */
+    public function profile(Request $request)
+    {
+        $currentUser = $request->user();
+
+        $user_profile = [
```
Comment (emugaya) [inline app/Http/Controllers/V1/UserController.php]: We don't have to save the user's `email, first_name` and `last_name`. These are always available to us in the token and user object passed when a request is made.

File: app/Http/Controllers/V1/UserController.php [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    use RESTActions;
+
+    /**
+     * Retrieve the user's profile.
+     *
+     * @param int $request - The request object.
+     *
+     * @return Response object.
+     */
+    public function profile(Request $request)
```
Comment (emugaya) [inline app/Http/Controllers/V1/UserController.php]: Rename this method to `getProfile` or any alternative name that's descriptive enough.

File: app/Http/Controllers/V1/UserController.php (hunk 3 above)
Comment (chadwalt) [reply app/Http/Controllers/V1/UserController.php]: I use RESTActions to have access to the `$this->respond()` method.

File: app/Http/Controllers/V1/UserController.php (hunk 5 above)
Comment (chadwalt) [reply app/Http/Controllers/V1/UserController.php]: This has been addressed.

File: app/Http/Controllers/V1/UserController.php (hunk 2 above)
Comment (chadwalt) [reply app/Http/Controllers/V1/UserController.php]: I have addressed this.

**Pull request merged**